MainModule/Server/Dependencies/Assets/Fly.lua

```diff
@@ -1,387 +1,87 @@
 end
-
-function getCF(part, isFor)
-	local cframe = part.CFrame
-	local noRot = CFrame.new(cframe.p)
-	local x, y, z = (workspace.CurrentCamera.CoordinateFrame - workspace.CurrentCamera.CoordinateFrame.p):toEulerAnglesXYZ()
-	return noRot * CFrame.Angles(isFor and z or x, y, z)
-end
-
-function dirToCom(part, mdir)
-	local dirs = {
-		Forward = ((getCF(part, true)*CFrame.new(0, 0, -1)) - part.CFrame.p).p;
-		Backward = ((getCF(part, true)*CFrame.new(0, 0, 1)) - part.CFrame.p).p;
-		Right = ((getCF(part)*CFrame.new(1, 0, 0)) - part.CFrame.p).p;
-		Left = ((getCF(part)*CFrame.new(-1, 0, 0)) - part.CFrame.p).p;
-	}
-	
-	for i,v in next,dirs do
-		if (v - mdir).magnitude <= 1.05 and mdir ~= Vector3.new(0,0,0) then
-			dir[i] = true
-		elseif not keyTab[i] then
-			dir[i] = false
+local Enabled=true
+UIS.InputBegan:Connect(function(Key,GC)
+	if GC then return end
+	if Key.KeyCode==Enum.KeyCode.E and not UIS:GetFocusedTextBox()then
+		if Ch.Parent and RootPart and Hum then
+			CF=RootPart.CFrame
+			Enabled=not Enabled
+			if Enabled then
+				Hum.PlatformStand = true
+			else
+				Hum.PlatformStand = false
+			end
 		end
 	end
-end
-
-function Start()
-	local curSpeed = 0
-	local topSpeed = speedVal.Value
-	local speedInc = topSpeed/25
-	local camera = workspace.CurrentCamera
-	local antiReLoop = {}
-	local realPos = part.CFrame
-	
-	speedVal.Changed:Connect(function()
-		topSpeed = speedVal.Value
-		speedInc = topSpeed/25
-	end)
-	
-	bPos.position = part.Position
-	bPos.maxForce = Vector3.new(math.huge, math.huge, math.huge)
-
-	bGyro.cframe = part.CFrame
-	bGyro.maxTorque = Vector3.new(9e9, 9e9, 9e9)
-	
-	antiLoop = antiReLoop
-	
-	if noclip.Value then
-		conn = runService.Stepped:Connect(function()
-			for _,v in pairs(char:GetDescendants()) do
-				if v:IsA("BasePart") then
-					v.CanCollide = false
+end)
+RunS.Heartbeat:Connect(function()
+	if Enabled and Ch.Parent and RootPart then
+		RootPart.Velocity = ZeroVector
+		RootPart.RotVelocity = ZeroVector
+		local Direction = UIS:GetFocusedTextBox()and ZeroVector or not UIS.KeyboardEnabled and Hum.MoveDirection or(ZeroVector +
+			(UIS:IsKeyDown'W'and Vector3.new(0, 0, -1)or ZeroVector) +
+			(UIS:IsKeyDown'S'and Vector3.new(0, 0, 1)or ZeroVector) +
+			(UIS:IsKeyDown'D'and Vector3.new(1, 0, 0)or ZeroVector) +
+			(UIS:IsKeyDown'A'and Vector3.new(-1, 0, 0)or ZeroVector) +
+			(UIS:IsKeyDown'Q'and Vector3.new(0, -1, 0)or ZeroVector) +
+			((UIS:IsKeyDown'E'or UIS:IsKeyDown(Enum.KeyCode.Space))and Vector3.new(0, 1, 0)or ZeroVector))
+		Direction = Direction*2*(UIS:IsKeyDown'LeftControl'and 3 or speedVal.Value)
+		CF = CF * CFrame.new(Direction)
+		local CDirection = (Mouse.Hit.Position-Camera.CFrame.Position)
+		CDirection = Camera.CFrame.Position + (CDirection.Unit * 10000)
+		CF = CFrame.new(CF.Position, not UIS.KeyboardEnabled and Camera.CFrame.Position or CDirection)
+		if noclip.Value then
+			for Indx, Obj in ipairs(Ch:GetChildren())do
+				if Obj:IsA"BasePart"then
+					Obj.CanCollide = false
```

Comment:
Please do not use Obj.CanCollide.
Please use https://developer.roblox.com/en-us/api-reference/class/NoCollisionConstraint instead.

How you would a complish this is with something like this.


local Evns, NoColConstraints= {}, {}
for _, v in ipairs({}) do
	if not Evns[v] then
		Evns[v] = v.Touched:Connect(function(Part)
			if Part.CanCollide = true and not NoColConstraints[Part] then
				local NoCol = Instance.new("NoCollisionConstraint")
				NoCol.Enabled = true
				NoCol.Part0 = v
				NoCol.Part1 = Part
				Debris:AddItem(NoCol, 1)
				NoColConstraints[Part] = NoCol
			end
		end)
	end
end

And stop flying either via hotkey or :unfly

for _, v in pairs(Evns) do
	v:Disconnect()
end
for _, v in ipairs(NoColConstraints) do
	v:Destroy()
end

Of course this was just a demonstration but you would run any of that code at hearbeat. Only run it on flight stop and start. Also store tables somewhere at the beginning of the script and define debris as local Debris = game:GetService("Debris")